src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/ConvertToJsonCommand.cs

```diff
@@ -25,8 +25,7 @@ public class ConvertToJsonCommand : PSCmdlet
         [AllowNull]
```

Comment:
@iSazonov, your last commit had 6 failures in `PowerShell-CI-macos`
(These are 5 of the failures)

TestConvertToJsonWithEnum
```powershell
System.Text.Json.JsonException : A possible object cycle was detected which is not supported. This can either be due to a cycle or if the object depth is larger than the maximum allowed depth of 1.
   at System.Text.Json.ThrowHelper.ThrowInvalidOperationException_SerializerCycleDetected(Int32 maxDepth)
   at System.Text.Json.JsonSerializer.Write(Utf8JsonWriter writer, Int32 originalWriterDepth, Int32 flushThreshold, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.JsonSerializer.WriteCore(Utf8JsonWriter writer, Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCore(PooledByteBufferWriter output, Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCoreString(Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Serialize(Object value, Type inputType, JsonSerializerOptions options)
   at Microsoft.PowerShell.Commands.JsonObject.ConvertToJson(Object objectToProcess, ConvertToJsonContext& context) in /Users/runner/runners/2.160.1/work/1/s/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs:line 505
   at PSTests.Parallel.UtilsTests.TestConvertToJsonWithEnum() in /Users/runner/runners/2.160.1/work/1/s/test/xUnit/csharp/test_Utils.cs:line 106
```
TestConvertToJsonBasic
```powershell
System.Text.Json.JsonException : A possible object cycle was detected which is not supported. This can either be due to a cycle or if the object depth is larger than the maximum allowed depth of 1.
   at System.Text.Json.ThrowHelper.ThrowInvalidOperationException_SerializerCycleDetected(Int32 maxDepth)
   at System.Text.Json.JsonSerializer.Write(Utf8JsonWriter writer, Int32 originalWriterDepth, Int32 flushThreshold, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.JsonSerializer.WriteCore(Utf8JsonWriter writer, Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCore(PooledByteBufferWriter output, Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCoreString(Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Serialize(Object value, Type inputType, JsonSerializerOptions options)
   at Microsoft.PowerShell.Commands.JsonObject.ConvertToJson(Object objectToProcess, ConvertToJsonContext& context) in /Users/runner/runners/2.160.1/work/1/s/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs:line 505
   at PSTests.Parallel.UtilsTests.TestConvertToJsonBasic() in /Users/runner/runners/2.160.1/work/1/s/test/xUnit/csharp/test_Utils.cs:line 89
```
TestConvertToJsonWithoutCompress
```powershell
System.Text.Json.JsonException : A possible object cycle was detected which is not supported. This can either be due to a cycle or if the object depth is larger than the maximum allowed depth of 1.
   at System.Text.Json.ThrowHelper.ThrowInvalidOperationException_SerializerCycleDetected(Int32 maxDepth)
   at System.Text.Json.JsonSerializer.Write(Utf8JsonWriter writer, Int32 originalWriterDepth, Int32 flushThreshold, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.JsonSerializer.WriteCore(Utf8JsonWriter writer, Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCore(PooledByteBufferWriter output, Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCoreString(Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Serialize(Object value, Type inputType, JsonSerializerOptions options)
   at Microsoft.PowerShell.Commands.JsonObject.ConvertToJson(Object objectToProcess, ConvertToJsonContext& context) in /Users/runner/runners/2.160.1/work/1/s/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs:line 505
   at PSTests.Parallel.UtilsTests.TestConvertToJsonWithoutCompress() in /Users/runner/runners/2.160.1/work/1/s/test/xUnit/csharp/test_Utils.cs:line 125
```
TestConvertToJsonCancellation
```powershell
System.Text.Json.JsonException : A possible object cycle was detected which is not supported. This can either be due to a cycle or if the object depth is larger than the maximum allowed depth of 1.
   at System.Text.Json.ThrowHelper.ThrowInvalidOperationException_SerializerCycleDetected(Int32 maxDepth)
   at System.Text.Json.JsonSerializer.Write(Utf8JsonWriter writer, Int32 originalWriterDepth, Int32 flushThreshold, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.JsonSerializer.WriteCore(Utf8JsonWriter writer, Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCore(PooledByteBufferWriter output, Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCoreString(Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Serialize(Object value, Type inputType, JsonSerializerOptions options)
   at Microsoft.PowerShell.Commands.JsonObject.ConvertToJson(Object objectToProcess, ConvertToJsonContext& context) in /Users/runner/runners/2.160.1/work/1/s/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs:line 505
   at PSTests.Parallel.UtilsTests.TestConvertToJsonCancellation() in /Users/runner/runners/2.160.1/work/1/s/test/xUnit/csharp/test_Utils.cs:line 146
```
Json Bug fixes.ConvertTo-Json -Depth 1001 throws MaximumAllowedDepthReached when the user specifies a depth greater than 1000.
```powershell
Expected an exception, with FullyQualifiedErrorId 'System.Text.Json.JsonException,Microsoft.PowerShell.Commands.ConvertToJsonCommand' to be thrown, but no exception was thrown.
at <ScriptBlock>, /Users/runner/runners/2.160.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Json.Tests.ps1: line 1410
1410:                 { $previous | ConvertTo-Json -Depth $testCase.MaxDepth } | Should -Throw -ErrorId $testCase.FullyQualifiedErrorId
```